ott/core/quad_problems.py

```diff
@@ -43,19 +43,32 @@ def marginal(self, axis: int = 0) -> jnp.ndarray:
     ...
 
 
-LossTerm = Callable[[jnp.ndarray], jnp.ndarray]
-Loss = Tuple[Tuple[LossTerm, LossTerm], Tuple[LossTerm, LossTerm]]
+class Loss(NamedTuple):
+  func: Callable[[jnp.ndarray], jnp.ndarray]
+  is_linear: bool
```

Comment:
would it make sense to set it to False by default?

Reply:
I want to keep this explicit so when new losses are added, one has to think about whether these terms are linear.